docs/developers/developer-reference/rust-testing-framework.md

```diff
@@ -0,0 +1,456 @@
+- caller address
+- contract wrapper (which contains the contract address and the contract object builder)
+- EGLD payment amount
+- a lambda function, which contains the actual execution
+
+Since this is a SC deploy, we call the `init` function. Since the contract works with managed objects, we can't use the built-in Rust BigUint. We have to use the one from elrond_wasm. To create managed types, we use the `managed_` functions. Alternatively, you can create those objects by:
+
+```rust
+let target = BigUint::<DebugApi>::from(2_000u32);
+```
+
+Similarly for other managed types.  
+
+The return type for the `execute_tx` function is an enum of two possible values: Commit and Revert. This is needed for cases when you want to check SCError cases (since the lambda can contain anything, the testing framework has no idea if your mocked transaction was successful or not).  
+
+After running the `init` function, we add a `setState` step in the generated mandos, to simulate our deploy.  
+
+To test the scenario and generate the mandos file, you have to create a test function:
+
+```rust
+#[test]
+fn init_test() {
+    let cf_setup = setup_crowdfunding(crowdfunding_esdt::contract_obj);
+    cf_setup
+        .blockchain_wrapper
+        .write_mandos_output("_generated_init.scen.json");
+}
+```
+
+And you're done for this step. You successfuly tested your contract's init function, and generated a mandos scenario for it.  
+
+## Testing transactions
+
+Let's test the `fund` function. For this, we're going to use the previous setup, and the `execute_esdt_transfer` method:
+
+```rust
+#[test]
+fn fund_test() {
+    let mut cf_setup = setup_crowdfunding(crowdfunding_esdt::contract_obj);
+    let b_wrapper = &mut cf_setup.blockchain_wrapper;
+    let user_addr = &cf_setup.first_user_address;
+
+    b_wrapper.execute_esdt_transfer(
+        user_addr,
+        &cf_setup.cf_wrapper,
+        CF_TOKEN_ID,
+        0,
+        &rust_biguint!(1_000),
+        |sc| {
+            let result = sc.fund(managed_token_id!(CF_TOKEN_ID), managed_biguint!(1_000));
+            assert_eq!(result, SCResult::Ok(()));
+
+            let user_deposit = sc.deposit(&managed_address!(user_addr)).get();
+            let expected_deposit = managed_biguint!(1_000);
+            assert_eq!(user_deposit, expected_deposit);
+
+            StateChange::Commit
+        },
+    );
+}
+```
+
+As you can see, we can directly call the storage mappers (like `deposit`) from inside the contract and compare with a local value. No need to encode anything.  
+
+Note: Even though we've already passed the payment in the `execute_esdt_transfer` method, we also have to pass it in the `fund` method of the smart contract. This is because, by calling the sc method directly, the `#[payment]` macros do not intervene yet, so the function argument is not auto-filled.  
```

Comment:
```suggestion
Note: Even though we've already passed the payment in the `execute_esdt_transfer` method, we also have to pass it in the `fund` method of the smart contract. This is because by calling the sc method directly, the `#[payment]` macros do not intervene as they would during a normal transaction, so the function argument is not auto-filled.  
```